pgex/examples/collision_shapes/main.py

```diff
@@ -0,0 +1,335 @@
+        ticks = pygame.time.get_ticks()
+
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                raise SystemExit(0)
+            
+        # Move shapes
+        circles[1].center = pygame.Vector2(90, 90) + pygame.Vector2(20).rotate(ticks / 3)
+        circles[2].center = pygame.Vector2(330, 90) + pygame.Vector2(20).rotate(ticks / 5)
+        rects[1].center = pygame.Vector2(205, 82.5) + pygame.Vector2(0, 40).rotate(ticks / 3)
+        polygons[0].angle = int(ticks / 10)
+        polygons[1].angle = -int(ticks / 10)
+        polygons[1].center.x = 350 + sin(ticks / 400) * 40
+
+        # This set contains references to collided shapes each frame for rendering
+        collided = []
+
+        # Check colliding shapes
+        # IMPORTANT: This is a really inefficient way if there are lots of shapes in the world.
+        #            Since this is just a showcase demo we are iteratting over all shapes.
+        for shape_a in all_shapes:
+            for shape_b in all_shapes:
+
+                # Early out if the two reference to same shape
+                if shape_a is shape_b: continue
+
+                # Circle and circle collision
+                if isinstance(shape_a, Circle) and isinstance(shape_b, Circle):
+                    if circle_x_circle(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Rect and circle collision
+                elif isinstance(shape_a, pygame.FRect) and isinstance(shape_b, Circle):
+                    if rect_x_circle(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Rect and rect collision
+                elif isinstance(shape_a, pygame.FRect) and isinstance(shape_b, pygame.FRect):
+                    if shape_a.colliderect(shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Polygon and circle collision
+                elif isinstance(shape_a, Polygon) and isinstance(shape_b, Circle):
+                    if polygon_x_circle(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Polygon and rect collision
+                elif isinstance(shape_a, Polygon) and isinstance(shape_b, pygame.FRect):
+                    if polygon_x_rect(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+                # Polygon and polygon collision
+                elif isinstance(shape_a, Polygon) and isinstance(shape_b, Polygon):
+                    if polygon_x_polygon(shape_a, shape_b):
+                        collided.append(shape_a)
+                        collided.append(shape_b)
+
+        # Render scene
+
+        window.fill((255, 255, 255))
```

Comment:
I would use constants for colors, so it is easy to update them across whole file (or even project) when we want to adjust them (say we want to have our red entity a bit darker), and it is easier for people to understand what they should expect to see on the screen while loading the application.

Easy to remember that` (255, 255, 255)` is white, but what is `(187, 255, 77)`? I have no idea without looking for it on Google.